explorer/testdata/interface/extends.carbon

```diff
@@ -0,0 +1,36 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// RUN: %{explorer} %s 2>&1 | %{FileCheck-strict} %s
+// RUN: %{explorer-trace} %s 2>&1 | %{FileCheck-allow-unmatched} %s
+// AUTOUPDATE: %{explorer} %s
+// CHECK:C.G
+// CHECK:result: 5
+
+package ExplorerTest api;
+
+interface A(T:! Type) {
+  fn F[me: Self]() -> T;
+}
+
+interface B {
+  extends A(i32);
+  fn G[me: Self]();
+}
+
+class C {
+  var n: i32;
+}
+
+// This impl also implements C as A(i32).
+external impl C as B {
+  fn F[me: Self]() -> i32 { return me.n; }
+  fn G[me: Self]() { Print("C.G"); }
+}
```

Comment:
Please use more meaningful names. I'd even be fine with A -> InterfaceA, B -> InterfaceB, F -> Func1, G -> Func2. But it's hard to quickly see correlations when you have a bunch of single-character names.

Reply:
Done.